File: RB110/study/Medium_2/triangles.rb
=begin


A triangle is classified as follows:

equilateral All 3 sides are of equal length
isosceles 2 sides are of equal length, while the 3rd is different
scalene All 3 sides are of different length
To be a valid triangle, the sum of the lengths of the two shortest sides must be greater than the length of the 
longest side, and all sides must have lengths greater than 0: if either of these conditions is not satisfied, 
the triangle is invalid.

Write a method that takes the lengths of the 3 sides of a triangle as arguments, 
and returns a symbol :equilateral, :isosceles, :scalene, or :invalid depending on whether the triangle 
is equilateral, isosceles, scalene, or invalid.

Examples:

Copy Code
triangle(3, 3, 3) == :equilateral
triangle(3, 3, 1.5) == :isosceles
triangle(3, 4, 5) == :scalene
triangle(0, 3, 3) == :invalid
triangle(3, 1, 1) == :invalid

Input:  
  -3 integers representing the sides of a triangle

Ouput:
  -A symbol representing what type of tringle is input 

Data structure
  -integers
  -symbols

Algorithm
  -Creat an array with the sides as elements
  -If All thre side are equal 
    -return :equilateral
  -If 2 sides are equal and the third not
    -return isosceles
  -If all sides are different lenth
    -return :scalene
  -If sum of min by 2 of the array is less than the third side
    -return invalid

=end


def triangle(side_1, side_2, side_3)
  tri_array = [side_1, side_2, side_3]
  shortest = [side_1, side_2, side_3].min(2).sum
  max = [side_1, side_2, side_3].max

  if shortest <= max
    :invalid
  elsif tri_array.all?(side_1)
    :equilateral
  elsif side_1 == side_2 || side_2 == side_3 || side_1 == side_3
    :isosceles
  elsif tri_array.uniq == tri_array
    :scalene
  
  end


end

  

p triangle(3, 3, 3) #== :equilateral
p triangle(3, 3, 1.5) #== :isosceles
p triangle(3, 4, 5) #== :scalene
p triangle(0, 3, 3) #== :invalid
p triangle(3, 1, 1) #== :invalid

File: LS215/exerises/medium2/triangles.js
/*
A triangle is classified as follows:

Equilateral: All three sides are of equal length.
Isosceles: Two sides are of equal length, while the third is different.
Scalene: All three sides are of different lengths.
To be a valid triangle, the sum of the lengths of the two shortest sides must be greater than the 
length of the longest side, and every side must have a length greater than 0. If either of these conditions 
is not satisfied, the triangle is invalid.

Write a function that takes the lengths of the three sides of a triangle as arguments and returns one of the 
following four strings representing the triangle's classification: 'equilateral', 'isosceles', 'scalene', or 
'invalid'.

Input:
  - 3 numbers that represents the sides of a triangle
Output:
  - A string that determines what type of triangle

Rules:
  - determine the type of triangle depending on the sides given
  - 

Questions:
  - will there always be 3 arguments
  - Will the arguments always be numbers?

Algorithm:
  - Determine if the sides given equal a valid triangle
    - Get the shortest side
    - Get the longest side
    - get the middle
      - If the shortest is less then or equal to 0
        - Return invalid
      - If the the shirtest and the middle is lest then the longest
        - Return invalid
    - If all the side are equal 
      - return "Equalateral"
    - If two sides are equal return 
      - Isosoles
    - If all the side are differen lengths
      - Retuen Salene

*/

function triangle(side1, side2, side3) {
  let sides =  [side1, side2, side3].sort();
  if (sides[0] <= 0) return 'invalid';
  if ((sides[1] + sides[0]) < sides[2]) return 'invalid';

  if (sides.every(side => sides[0] === side)) return 'equilateral';
  if (sides[0] !== sides[1] && sides[1] !== sides[2] && sides[0] !== sides[2]) return 'scalene'
  if (sides.some(side => side === sides[0])) return 'isosceles';
  
}


console.log(triangle(3, 3, 3));        // "equilateral"
console.log(triangle(3, 3, 1.5));      // "isosceles"
console.log(triangle(3, 4, 5));        // "scalene"
console.log(triangle(0, 3, 3));        // "invalid"
console.log(triangle(3, 1, 1));        // "invalid"